server/battles/BattleResultProcessor.cpp

```diff
@@ -487,6 +487,16 @@ void BattleResultProcessor::endBattleConfirm(const CBattleInfoCallback & battle)
 	// Remove beaten hero
 	if(finishingBattle->loserHero)
 	{
+		if(!finishingBattle->isDraw())
+		{
+			ConstTransitivePtr<CGHeroInstance> strongestHero = nullptr;
+			for(auto & hero : gameHandler->gameState()->getPlayerState(finishingBattle->loser)->heroes)
+				if(!strongestHero || hero->exp > strongestHero->exp)
+					strongestHero = hero;
+			if(strongestHero->id == finishingBattle->loserHero->id)
+				gameHandler->gameState()->statistic.accumulatedValues[finishingBattle->victor].lastDefeatedStrongestHeroDay = gameHandler->gameState()->getDate(Date::DAY);
+		}
+
```

Comment:
Might be better to move code like this into GameStatistics class.
For example, add method `onBattleFinished` that would collect any stats related to end of battle

Reply:
Yeah, but not as part as this PR, I think. Needs refactoring. `FinishingBattleHelper` is a server class `GameStatistics` is lib. This PR is already big.